modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/inputtype/text/TextInputType.ts

```diff
@@ -0,0 +1,78 @@
+module api.form.inputtype.text {
+
+    import support = api.form.inputtype.support;
+    import Property = api.data.Property;
+    import Value = api.data.Value;
+    import ValueType = api.data.ValueType;
+    import ValueTypes = api.data.ValueTypes;
+    import i18n = api.util.i18n;
+    import NumberHelper = api.util.NumberHelper;
+    import DivEl = api.dom.DivEl;
+    import FormInputEl = api.dom.FormInputEl;
+    import Element = api.dom.Element;
+
+    export abstract class TextInputType extends support.BaseInputTypeNotManagingAdd<string> {
+
+        private maxLength: number;
+
+        constructor(config: api.form.inputtype.InputTypeViewContext) {
+            super(config);
+            this.readConfig(config.inputConfig);
+        }
+
+        protected readConfig(inputConfig: { [element: string]: { [name: string]: string }[]; }): void {
+            const maxLengthConfig = inputConfig['maxLength'] ? inputConfig['maxLength'][0] : {};
+            const maxLength = NumberHelper.toNumber(maxLengthConfig['value']);
+            this.maxLength = maxLength > 0 ? maxLength : null;
+        }
+
+        protected initOccurenceListeners(inputEl: FormInputEl) {
+
+            inputEl.onValueChanged((event: api.ValueChangedEvent) => {
+                if (NumberHelper.isNumber(this.maxLength)) {
```

Comment:
Value in ``this.maxLength`` will be set only once and never change, so there's no point in checking for ``isNumber`` every time inside these two handlers. Instead, we should create both event handlers only **if** ``this.maxLength`` is a number:

```Typescript
if (NumberHelper.isNumber(this.maxLength)) {
                inputEl.onValueChanged((event: api.ValueChangedEvent) => {
                    const lengthCounter = Element.fromHtmlElement(
                        (<HTMLElement>inputEl.getParentElement().getHTMLElement().querySelector('.length-counter')));
                    if (lengthCounter) {
                        this.updateLengthCounterValue(lengthCounter, inputEl.getValue());
                    }
                });

                inputEl.onRendered(() => {
                    const lengthCounter = new DivEl('length-counter');
                    this.updateLengthCounterValue(lengthCounter, inputEl.getValue());

                    inputEl.getParentElement().appendChild(lengthCounter);
                });
            }
```